Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/NodeGraph.cpp

```diff
@@ -231,31 +231,31 @@ namespace EMStudio
                 // add the playspeed
                 if (plugin->GetIsDisplayFlagEnabled(AnimGraphPlugin::DISPLAYFLAG_PLAYSPEED))
                 {
-                    m_qtTempString.asprintf("Play Speed = %.2f", emfxNode->GetPlaySpeed(animGraphInstance));
+                    m_qtTempString = AZStd::string::format("Play Speed = %.2f", emfxNode->GetPlaySpeed(animGraphInstance)).c_str();
```

Comment:
Small string optimization is currently 16 characters in `AZStd::string` so if `GetPlaySpeed` is longer more than 2 digits the highligted string is going to have to allocate memory, so in it's really close to that particular edge. Besides that, since `m_qtTempString` is a QT string, it will need to allocate memory regardless, but QString I believe uses lazy copies so using the formatter should technically only allocate memory once which would have happened anyway.
As for using `QString::asprintf()` instead of `QString().arg()`, the former is being phased out and the latter is the preferred approach these days, but I'm not a QT expert by any stretch of the imagination so I'll let more experienced contributors chime in on that one.

Reply:
Oh, did that get reduced to 16? My last info was that it is around 30ish?

@NicholasVanSickle what's your take? Do you see a way here without an allocation?

We're running this per anim graph node, per UI update tick, so it will result in quite a few allocations if we don't have a way to do that without one.

What about using sprintf with a buffer on the stack?

Reply:
Could you potentially use `AZStd::fixed_string` here? Might be a good use case to avoid a potential allocation?

Reply:
That sounds like a good idea to me @hultonha, Thanks! Everyone else on-board with that?

Reply:
Changed, please check the updated version.